server/controllers/eventController.ts

```diff
@@ -0,0 +1,200 @@
+import { Response, Request, NextFunction } from "express";
+import AppError from "../utils/AppError";
+import { PrismaClient } from "@prisma/client";
+import { sendMail } from "../utils/Email";
+
+const prisma = new PrismaClient();
+
+// get all events
+export const getAllEvents = async (req: Request, res: Response, next: NextFunction) => {
+    try {
+        const events = await prisma.event.findMany();
+
+        if(events.length < 1){
+            return next(new AppError("No events found", 404));
+        }
+
+        res.status(200).json({
+            status: "success",
+            data: events,
+        });
+    } catch (error) {
+        next(new AppError("Failed to get events", 500));
+    }
+};
+
+//get single event
+export const getEvent = async (req: Request, res: Response, next: NextFunction) => {
+    try {
+        const { id } = req.params;
+
+        const event = await prisma.event.findUnique({
+            where: { 
+                id: parseInt(id) 
+            },
+            include:{
+                createdBy: {
+                    select: {
+                        name: true,
+                        email: true,
+                        imageUrl: true
+                    },
+                },
+                collaborators: {
+                    select: {
+                        user: {
+                            select: {
+                                name: true,
+                                email: true,
+                                imageUrl: true
+                            },
+                        },
+                        role: true
+                    },
+                },
+                editSessions: true
+            }
+        });
+
+        if(!event){
+            return next(new AppError("Event not found", 404));
+        }
+
+        res.status(200).json({
+            status: 'success',
+            data: event,
+        });
+    } catch (error) {
+        console.log("Error getting event", error);
+        return next(new AppError("Failed to get event", 500));
+    }
+};
+
+// create new event
+export const createEvent = async (req: Request, res: Response, next: NextFunction) => {
+    try {
+        const { title, description, startTime, endTime, createdById, collaborators, editSessions } = req.body;
+
```

Comment:
check if the token starts with 'Bearer' and the splits it to 2 strings and access the second string which is our token.